ocaml/xenopsd/lib/xenops_server.ml

```diff
@@ -2398,6 +2398,13 @@ and perform_exn ?subtask ?result (op : operation) (t : Xenops_task.task_handle)
               fun fd fn -> fn fd
           (* do nothing *)
         in
+        let compress_vgpu vgpu_fd f =
+          match vgpu_fd with
+          | Some (FD fd) when compress_memory ->
+              compress fd (fun fd -> f (Some (FD fd)))
```

Comment:
`compress` already becomes the identity function if `compress_memory = false`, so can we just use `compress` directly?

Reply:
We can't use `compress` here because the argument type is not a `fd` but more complicated and needs to be handled.